code/core/src/common/utils/file-cache.ts

```diff
@@ -1,11 +1,157 @@
-import * as fsc from 'file-system-cache';
+import { createHash, randomBytes } from 'node:crypto';
```

Comment:
let's put a big fat comment here, explaining that this is inlined from the original package (and when/which version), giving attribution to the original author.